tests/test_update_version_next.py

```diff
@@ -0,0 +1,90 @@
+"""Tests for the update_version_next tool."""
+
+import unittest
+from pathlib import Path
+
+# typeshed doesn't have stubs for os_helper
+from test.support import os_helper  # type: ignore[import-not-found]
+
+import update_version_next
+
+TO_CHANGE = """
+Directives to change
+--------------------
+
+Here, all occurences of NEXT (lowercase) should be changed:
+
+.. versionadded:: next
+
+.. versionchanged:: next
+
+.. deprecated:: next
+
+.. deprecated-removed:: next 4.0
+
+whitespace:
+
+..   versionchanged:: next
+
+.. versionchanged  :: next
+
+    .. versionadded:: next
+
+arguments:
+
+.. versionadded:: next
+    Foo bar
+
+.. versionadded:: next as ``previousname``
+"""
+
+UNCHANGED = """
+Unchanged
+---------
+
+Here, the word "next" should NOT be changed:
+
+.. versionchanged:: NEXT
+
+..versionchanged:: NEXT
+
+... versionchanged:: next
+
+foo .. versionchanged:: next
+
+.. otherdirective:: next
+
+.. VERSIONCHANGED: next
+
+.. deprecated-removed: 3.0 next
+"""
+
+EXPECTED_CHANGED = TO_CHANGE.replace("next", "VER")
+
+
+class TestVersionNext(unittest.TestCase):
```

Comment:
We can use pytest-style tests here with class-less functions and plain asserts.

Reply:
And magic fixtures, right? :)